enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverter.java

```diff
@@ -130,15 +131,18 @@ private FormData prepareFormData(
         }
 
         final String attributes = operationTemplate.getAttributes();
-        if (attributes != null) {
-            final OperationTemplateParam[] operationTemplateParams = objectMapper.readValue(attributes, OperationTemplateParam[].class);
-            if (operationTemplateParams != null) {
-                final List<Attribute> formDataAttributes = formData.getAttributes();
-                for (OperationTemplateParam templateParam : operationTemplateParams) {
-                    buildAttribute(templateParam, parameters)
-                            .ifPresent(formDataAttributes::add);
-                }
-            }
+        if (attributes == null) {
+            return formData;
+        }
+
+        final List<OperationTemplateParam> operationTemplateParams = objectMapper.readValue(attributes, new TypeReference<>(){});
+        if (operationTemplateParams != null) {
+            final List<Attribute> formDataAttributes = operationTemplateParams.stream()
+                    .map(templateParam -> buildAttribute(templateParam, parameters))
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
```

Comment:
This could be one line shorter when using `.flatMap(Optional::stream)`.

Reply:
🤔 Nice!